packages/jellyfish-api-core/__tests__/category/loan/updateLoanScheme.test.ts

```diff
@@ -0,0 +1,185 @@
+      const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+      expect(result.length).toStrictEqual(1)
+      expect(result[0]).toStrictEqual(
+        {
+          default: false,
+          id: 'scheme1',
+          interestrate: 3.5,
+          mincolratio: 300
+        }
+      )
+    }
+  })
+
+  it('should not updateLoanScheme if activateAfterBlock is less than current block', async () => {
+    // Wait for block 130
+    await testing.container.waitForBlockHeight(130)
+
+    // Attempt to updateLoanScheme at block 129
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme1', activateAfterBlock: 129 })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nUpdate height below current block height, set future height\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if same minColRatio and interestRate pending loan scheme created before', async () => {
+    // Wait for block 140
+    await testing.container.waitForBlockHeight(140)
+
+    await testing.container.call('createloanscheme', [300, new BigNumber(3.5), 'scheme2'])
+    await testing.generate(1)
+
+    // To update scheme on later block
+    await testing.rpc.loan.updateLoanScheme(400, new BigNumber(4.5), { id: 'scheme2', activateAfterBlock: 150 })
+    await testing.generate(1)
+
+    // Attempt to update same minColRatio and interestRate as the pending scheme
+    const promise = testing.rpc.loan.updateLoanScheme(400, new BigNumber(4.5), { id: 'scheme1' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme scheme2 with same interestrate and mincolratio pending on block 150\', code: -32600, method: updateloanscheme')
+
+    // Update at block 150
+    await testing.container.waitForBlockHeight(150)
+  })
+
+  it('should updateLoanScheme with utxos', async () => {
+    const { txid, vout } = await testing.container.fundAddress(GenesisKeys[0].owner.address, 10)
+    const loanId = await testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme1', utxos: [{ txid, vout }] })
+    expect(typeof loanId).toStrictEqual('string')
+    expect(loanId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloanschemes')
+    const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+    expect(result.length).toStrictEqual(1)
+    expect(result[0]).toStrictEqual(
+      {
+        default: false,
+        id: 'scheme1',
+        interestrate: 3.5,
+        mincolratio: 300
+      }
+    )
+  })
+
+  it('should not updateLoanScheme  with utxos not from foundation member', async () => {
+    const { txid, vout } = await testing.container.fundAddress(await testing.generateAddress(), 10)
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme1', utxos: [{ txid, vout }] })
```

Comment:
```suggestion
    const utxo = await testing.container.fundAddress(await testing.generateAddress(), 10)
    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme1', utxos: [utxo] })
```

Reply:
i don't change for this. I add more test code that required txid and vout
```
    const rawtx = await testing.container.call('getrawtransaction', [loanSchemeId, true])
    expect(rawtx.vin[0].txid).toStrictEqual(txid)
    expect(rawtx.vin[0].vout).toStrictEqual(vout)
```